app/src/components/Calendar.js

```diff
@@ -1,61 +1,148 @@
 import React, { useState, useEffect } from "react";
 
+import dayjs from "dayjs";
+
 import Day from "./Day";
+import { CalendarNavBar } from "./NavBar";
 import TimeLine from "./TimeLine";
 
-function Calendar({ isAuthenticated, gcal }) {
-  const [dayStart, setDayStart] = useState(new Date());
-  const [dayEnd, setDayEnd] = useState(new Date());
+function Calendar({ isAuthenticated, gcal, timedownAccount }) {
+  const [isReady, setIsReady] = useState(false);
+  //Number of days the user wants to see
+  const [calView, setCalView] = useState("1day");
+  //Starting hour for day rendering
+  const [dayStart, setDayStart] = useState(dayjs().hour(0));
+  //Ending hour for day rendering
+  const [dayEnd, setDayEnd] = useState(dayjs().hour(23));
+  //Total hours rendered in a day
   const [totalHours, setTotalHours] = useState(24);
+  //Days to be displayed in calendar
+  const [days, setDays] = useState([
+    { start: dayjs().hour(0), end: dayjs().hour(23) },
+  ]);
+  // Adjust days state when user clicks "previous" or "next" in navigation
+  const [dateNavigation, setDateNavigation] = useState(0);
+  //Sets Calendar's columns based on calView
+  const [dayColumns, setDayColumns] = useState("1fr 4fr");
+  //Sets the rows for TimeLine and Day components, based on total hours
   const [timeRows, setTimeRows] = useState("");
 
-  //sets Date states for rendering
+  //sets dayStart, dayEnd, and totalHours states for rendering
   function setTimeRanges() {
-    //Dummy data for rendering purposes, will be replaced with functions that read
-    //from the user's settings and render the hours where the user is awake.
-    let start = new Date();
-    start.setHours(9, 0, 0, 0);
-    let end = new Date();
-    end.setHours(21, 0, 0, 0);
-
-    setDayStart(start);
-    setDayEnd(end);
-
-    let total = end.getHours() - start.getHours();
-    // console.debug({ total }); //TEST
-    setTotalHours(total);
+    if (timedownAccount.sleepTime) {
+      let start = dayjs().hour(timedownAccount.sleepTime.start.hours);
+      let end = dayjs().hour(timedownAccount.sleepTime.end.hours);
+
+      setDayStart(start);
+      setDayEnd(end);
+
+      let total = dayjs(end).hour() - dayjs(start).hour();
+      // console.debug({ total }); //TEST
+      setTotalHours(total);
+    }
```

Comment:
so cool that you added this customizable feature that pull from the user's settings! very thoughtful and considerate that people wake up and sleep at different times